build/pause/Makefile

```diff
@@ -12,70 +12,78 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-.PHONY: all push container clean orphan all-push push-manifest
-
-include ../../hack/make-rules/Makefile.manifest
+.PHONY: all push container clean orphan all-push push-manifest annotate manifest
 
 REGISTRY ?= staging-k8s.gcr.io
 IMAGE = $(REGISTRY)/pause
-IMAGE_WITH_ARCH = $(IMAGE)-$(ARCH)
+IMAGE_WITH_OS_ARCH = $(IMAGE)-$(OS)-$(ARCH)
 
 TAG = 3.1
 REV = $(shell git describe --contains --always --match='v*')
 
 # Architectures supported: amd64, arm, arm64, ppc64le and s390x
 ARCH ?= amd64
+# Operating systems supported: linux, windows
+OS ?= linux
 
-ALL_ARCH = amd64 arm arm64 ppc64le s390x
+ALL_OS = linux windows
+ALL_ARCH.linux = amd64 arm arm64 ppc64le s390x
+ALL_ARCH.windows = amd64
+ALL_OS_ARCH = $(foreach os, $(ALL_OS), $(foreach arch, ${ALL_ARCH.${os}}, $(os)-$(arch)))
 
 CFLAGS = -Os -Wall -Werror -static -DVERSION=v$(TAG)-$(REV)
-KUBE_CROSS_IMAGE ?= k8s.gcr.io/kube-cross
-KUBE_CROSS_VERSION ?= $(shell cat ../build-image/cross/VERSION)
+KUBE_CROSS_IMAGE.linux ?= k8s.gcr.io/kube-cross
+KUBE_CROSS_VERSION.linux ?= $(shell cat ../build-image/cross/VERSION)
+KUBE_CROSS_IMAGE.windows ?= dockcross/windows-static-x64
+KUBE_CROSS_VERSION.windows ?= latest
+KUBE_CROSS_IMAGE := ${KUBE_CROSS_IMAGE.${OS}}
+KUBE_CROSS_VERSION := ${KUBE_CROSS_VERSION.${OS}}
 
 BIN = pause
-SRCS = pause.c
-
-ifeq ($(ARCH),amd64)
-	TRIPLE ?= x86_64-linux-gnu
-endif
+SRCS.linux = linux/pause.c
+SRCS.windows = windows/pause.c
+SRCS := ${SRCS.${OS}}
 
-ifeq ($(ARCH),arm)
-	TRIPLE ?= arm-linux-gnueabihf
-endif
+# The manifest command is still experimental as of Docker 18.09.3
+export DOCKER_CLI_EXPERIMENTAL=enabled
 
-ifeq ($(ARCH),arm64)
-	TRIPLE ?= aarch64-linux-gnu
-endif
+TRIPLE.linux-amd64 := x86_64-linux-gnu
+TRIPLE.windows-amd64 := x86_64-w64-mingw32.static
+TRIPLE.linux-arm := arm-linux-gnueabihf
+TRIPLE.linux-arm64 := aarch64-linux-gnu
+TRIPLE.linux-ppc64le := powerpc64le-linux-gnu
+TRIPLE.linux-s390x := s390x-linux-gnu
+TRIPLE := ${TRIPLE.${OS}-${ARCH}}
 
-ifeq ($(ARCH),ppc64le)
-	TRIPLE ?= powerpc64le-linux-gnu
-endif
-
-ifeq ($(ARCH),s390x)
-	TRIPLE ?= s390x-linux-gnu
-endif
+BASE.linux := scratch
+BASE.windows := mcr.microsoft.com/windows/nanoserver:1809
```

Comment:
I think we should plan for `BASE.windows.version` at this point. `1809` (also known as 2019, which is the same bits but under LTS) is the only version supported now, but there will be new Windows versions shipping twice per year that we'll support. Multi-arch manifests can handle multiple versions already - see `docker manifest inspect mcr.microsoft.com/k8s/core/pause:1.0.0`